lib/connectors/memory.js

```diff
@@ -210,6 +210,11 @@ Memory.prototype.create = function create(model, data, callback) {
   if(!this.collection(model)) {
     this.collection(model, {});
   }
+
+  if (this.collection(model)[id]) {
+    return callback(new Error('Duplicate entry for ' + model + '.' + idName));
```

Comment:
We should make the callback inside `process.nextTick` so that the async expectation is guaranteed.